pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -73,6 +77,12 @@ type ResolvedPipelineRunTask struct {
 	ResolvedTaskResources *resources.ResolvedTaskResources
 	// ConditionChecks ~~TaskRuns but for evaling conditions
 	ResolvedConditionChecks TaskConditionCheckState // Could also be a TaskRun or maybe just a Pod?
+	// IsFailurePermitted is driven by the runOn section of the dependent tasks, runOn:  state: ["failure"]
+	// it is set to true if any of the tasks within the pipeline has dependency on this task's failure state
+	// by default, IsFailurePermitted is set to false
+	IsFailurePermitted bool
+	// IsRunOnConflicted is set to true for a task with runOn states conflicting with actual states of parent task
+	IsRunOnConflicted bool
```

Comment:
I'm not completely convinced that we need to store this state here. 
Could the function that returns the list of schedulable TaskRuns use the filters defined in runOn and the state of TaskRuns executed until now to decide if a taskrun should be schedulable or not?

NIT: the "conflicted" term does not resonate with me, but I'm miserably failing to offer an alternative that sounds good. Perhaps using the opposite - `isRunOnSatisfied` - but this does not sound very good to me either.

Reply:
> I'm not completely convinced that we need to store this state here.
> Could the function that returns the list of schedulable TaskRuns use the filters defined in runOn and the state of TaskRuns executed until now to decide if a taskrun should be schedulable or not?

Yup, that's exactly how its implemented. A taskrun should be scheduled or not is determined solely using the filters (states) defined in `runOn` and actual states of taskruns.
These flags are mainly used to determine (1) create skipped state taskrun if `IsRunOnConflicted` is set (2) pipelinerun decides whether to continue when a task fails if `IsFailurePermitted` is set for that task. Note here with DAG, a list of schedulable tasks is determined by looking at the N-1 level of the tree whereas `IsFailurePermitted` is set for a parent task based on its children `runOn` states.  

> NIT: the "conflicted" term does not resonate with me, but I'm miserably failing to offer an alternative that sounds good. Perhaps using the opposite - `isRunOnSatisfied` - but this does not sound very good to me either.

Haha 😂I am having same issue 😫, cannot come up with better option to signify conflict between a state specified in `runOn` vs actual taskrun state.